apps/vmq_server/src/vmq_mqtt5_fsm.erl

```diff
@@ -488,6 +488,9 @@ connected(#mqtt5_pubrec{message_id=MessageId, reason_code=RC}, State) when RC <
             _ = vmq_metrics:incr({?MQTT5_PUBREL_SENT, ?SUCCESS}),
             {State#state{waiting_acks=maps:update(MessageId, PubRelFrame, WAcks)},
              [serialise_frame(PubRelFrame)]};
+        #mqtt5_pubrel{message_id=MessageId} = PubRelFrame ->
+            _ = vmq_metrics:incr({?MQTT5_PUBREL_SENT, ?SUCCESS}),
+            {State, [serialise_frame(PubRelFrame)]};
```

Comment:
Well, then we'd also have to change the handling of the `#mqtt5_pubcomp` and I like it better that we there match against a `#mqtt5_pubrel` than against a generic `#vmq_msg{}`...

Reply:
Agree on this... but I guess we should add some documentation here that documents this edgecase of repeated PUBRECs... I guess they are repeated because VerneMQ has sent a dup PUBLISH before, right?

Reply:
Hmm, but is it an edge case? It's clearly stated in the spec that we *MUST* handle these repeated PUBRECs within an ongoing QoS2 flow. Not sure why this case appears, maybe just a client that retries too fast? It wasn't obvious to me that VerneMQ had retried and sent a dup PUBLISH, but it could be the case.